webboot/webboot.go

```diff
@@ -26,43 +33,134 @@ var (
+		if err != nil {
+			return "", "", fmt.Errorf("Invalid URL: got %v, want a valid URL. Got %v error, want nil", arg, err)
+		}
+	}
+	return link, filename, nil
+}
+
+// linkOpen returns an io.ReadCloser that holds the content of the OS
+func linkOpen(link string) (io.ReadCloser, error) {
+	log.Println("Retrieving the file")
+	resp, err := http.Get(link)
+	if err != nil {
+		return nil, err
+	}
+	// checks if request to receive URL is successful
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("non-200 HTTP status: %d", resp.StatusCode)
+	}
+	return resp.Body, nil
+}
+
+// write copies the content from the downloaded ISO into a new file with filename.
+func write(read io.ReadCloser, filename string) error {
+	/*
+		if err := os.MkdirAll(dir, 0660); err != nil {
+			return fmt.Errorf("Error making directory: got %v, want nil", err)
+		}
+	*/
+	//filename = filepath.Join(dir, filename)
+	w, err := os.Create(filename)
+	if err != nil {
+		return fmt.Errorf("Error creating a file: got %v, want nil", err)
+	}
+
+	defer w.Close()
+	defer read.Close()
+
+	if _, err := io.Copy(w, read); err != nil {
+		return fmt.Errorf("Error copying content into %v file, got %v, want nil", filename, err)
+	}
+	return nil
+}
+
+// name takes a link & generates a filename from it
+// For example, if a valid link = http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso, then filename = CorePure64-10.1.iso
+// if the link is empty or if the link's Path ends in /, name returns a default index.html as the filename
+func name(link string) (string, error) {
+	//parses through the URL user provides to generate a new name for filename that will hold the contents of the downloaded OS
+	p, err := url.Parse(link)
+
+	if err != nil {
+		return "", err
+	}
+	filename := "index.html"
+
+	if p.Path != "" && !strings.HasSuffix(p.Path, "/") {
+		filename = path.Base(p.Path)
+	}
+	return filename, nil
+}
+
+func usage() {
+	log.Printf("Usage: %s [ARGS] URL or name of ISO\n", os.Args[0])
+	flag.PrintDefaults()
+	os.Exit(1)
+}
+
 func main() {
 
 	flag.Parse()
 
-	dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
-
-	if len(flag.Args()) < 1 {
-		log.Fatal("error:pass in an operating system")
+	if flag.NArg() != 1 {
+		usage()
 	}
+	//Search to connect to one or more network interfaces
+	dhclient.Request(*ifName, *timeout, *retry, *verbose, *ipv4, *ipv6)
 
-	osname := flag.Args()[0]
+	//converts the argument to be lower case, so it is processed interally in an identical manner
+	//no matter if the user types "webboot Tinycore", "webboot TINYCORE", "webboot tinycore", etc.
+	arg := strings.ToLower(flag.Arg(0))
 
-	if _, ok := osSystems[osname]; !ok {
-		log.Printf("error operating system is not supported,The following systems are supported")
-		for os := range osSystems {
-			log.Printf("%v", os)
-		}
-		log.Fatalf("exit")
+	//Parses the user arg and splits it into different parts for webboot: link, filename
+	link, filename, err := parseArgs(arg)
```

Comment:
I changed the function name to parseArg() because I see your point on how it only takes one argument and parses it to see what to do with the one arg. I also got rid of the comment since the code is clear.